File: Java/src/main/java/com/gildedrose/categories/DefaultCategory.java
package com.gildedrose.categories;

import com.gildedrose.*;

public class DefaultCategory implements ItemCategory {

    public String getName() {
        return "";
    }

    public void UpdateQuality(Item item) {
        int factor = item.sellIn < 0 ? 2 : 1;

        item.quality -= factor;

        if (item.quality < 0) {
            item.quality = 0;
        }
    }

    public void UpdateSellIn(Item item) {
        item.sellIn--;
    }
}

File: csharpcore/Categories/DefaultCategory.cs
namespace csharpcore.Categories
{
    public class DefaultCategory : IItemCategory
    {
        public virtual string Name => string.Empty;

        public virtual void UpdateQuality(Item item)
        {
            var factor = item.SellIn < 0 ? 2 : 1;

            item.Quality -= factor;

            if (item.Quality < 0)
            {
                item.Quality = 0;
            }
        }

        public virtual void UpdateSellIn(Item item)
        {
            item.SellIn--;
        }
    }
}